sql/src/main/java/io/crate/expression/operator/CIDROperator.java

```diff
@@ -0,0 +1,158 @@
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.operator;
+
+import io.crate.data.Input;
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.functions.params.Param;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.network.InetAddresses;
+
+import java.math.BigInteger;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.BiFunction;
+
+public final class CIDROperator {
+
+    public static final String CONTAINED_WITHIN = Operator.PREFIX + "<<";
+
+    private static final int IPV4_ADDRESS_LEN = 4;
+    private static final int IPV6_ADDRESS_LEN = 16;
+
+    public static void register(OperatorModule module) {
+        module.registerDynamicOperatorFunction(CONTAINED_WITHIN, new CIDRResolver(CIDROperator::llt));
+    }
+
+    public static boolean llt(String ipStr, String cidr) {
+        if (null == ipStr || null == cidr) {
+            throw new IllegalArgumentException("operands cannot be null");
+        }
+        if (false == cidr.contains("/") || ipStr.contains("/")) {
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                "operands are incorrect, expected [ip, cidr], got [%s, %s]", ipStr, cidr));
+        }
+        try {
+            BigInteger ip = new BigInteger(1, InetAddress.getByName(ipStr).getAddress());
+            return extractRange(cidr, (start, end) -> start.compareTo(ip) <= 0 && ip.compareTo(end) < 0);
+        } catch (UnknownHostException uhe) {
+            throw new IllegalArgumentException(uhe);
+        }
+    }
+
+    public static Tuple<InetAddress, InetAddress> extractRange(String cidr) {
+        return extractRange(cidr, (start, end) -> new Tuple<>(fromBigIntegerIp(start), fromBigIntegerIp(end)));
+    }
+
+    @FunctionalInterface
+    private interface IpRangeConsumer<R> {
+        R accept(BigInteger ipStartRange, BigInteger ipEndRange);
+    }
+
+    private static <R> R extractRange(String cidr, IpRangeConsumer<R> rangeConsumer) {
+        if (null == cidr || false == cidr.contains("/")) {
+            throw new IllegalArgumentException(String.format(
+                Locale.ENGLISH, "operand [%s] must conform with CIDR notation", cidr));
+        }
```

Comment:
Maybe turn this into an assertion since the function is private and the contract is already verified in the public method as well.

Reply:
This query:

```@Test
    public void test_ipv4_cidr_operator_right_operand_is_ip() throws Throwable {
        // operand [192.168.1.0] must conform with CIDR notation
        expectedException.expect(IllegalArgumentException.class);
        test(COLUMN_NAME,
             new String[]{"192.168.0.1", "192.168.1.1", "192.168.1.7", "192.168.1.255", "192.168.2.0", "10.0.0.1"},
             String.format("%s << '192.168.1.0'::ip", COLUMN_NAME),
             "192.168.1.1", "192.168.1.7");
    }
```

results in the correct signature (ip, text) with a cidr lacking the prefix part, so an exception is thrown, AssertionError if I perform the replacement